index.js

```diff
@@ -215,10 +187,23 @@ DecryptionStream.prototype._transform = function(chunk, enc, cb) {
   // We can't use an else because we have no idea how long our chunks will be
   // all we know is that once we've got a nonce and mac decryption can begin
   if (this._started) {
-    this.push(this._decipher.update(chunk));
+    this._cipherTextChunks.push(chunk);
   }
 
-  if (typeof(cb) === 'function') {
-    cb();
+
+  cb();
+};
+
+DecryptionStream.prototype._flush = function(cb) {
+  var data = Buffer.concat(this._cipherTextChunks);// this could be rewritten to avoid doing this
+  var mac = data.slice(-16);
```

Comment:
we could be a bit smarter here though in how we do this, I can add something to the pull